### PR TITLE
024 standard_no hash, oclc, lccn, isbn, catalog date

### PR DESCRIPTION
**Pull request by tampakis**
- Adds solr field for [MARC 024](http://www.loc.gov/marc/bibliographic/bd024.html). Closes pulibrary/orangelight#22.
- Creates display and normalized fields for oclc, lccn, and isbn. Closes pulibrary/orangelight#223
- Indexes new 959 catalog date from MARC. Closes pulibrary/orangelight#117

### CODE REVIEW

File: lib/traject_config.rb [hunk 1]
```diff
@@ -696,7 +702,41 @@
...
+  accumulator.each_with_index do |value, i|
+    accumulator[i] = ISBN.from_string(value).gsub(/\s|\-/,'')
+  end
+end
+
+to_field 'oclc_param', extract_marc('035a') do |record, accumulator|
+  oclcs = []
+  accumulator.each_with_index do |value, i|
+    oclcs << value.gsub(/\D/, '') if value.start_with?('(OCoLC)')
+  end
```
Comment (kevinreiss) [inline lib/traject_config.rb:737]: Is (OCoLC) the only OCLC prefix style we use in our system? There have been a bunch of iterations over the years.

Comment (tampakis) [inline lib/traject_config.rb:737]: @joycebcat will OCLC numbers appear outside of the `035 $a(OCoLC)` context that I should include in this field?

Comment (joycebcat) [inline lib/traject_config.rb:737]: @tampakis No, I don't think so, and I've hunted around different ages of records and I can't find any examples which fall outside that context.

Comment (tampakis) [inline lib/traject_config.rb:737]: Thanks @joycebcat! What I did forget to catch is removing the `ocn` prefix and make sure there aren't duplicates. Right now the field returns:

```
 "ocn882089266",
 "882089266"
```

Comment (joycebcat) [inline lib/traject_config.rb:737]: @tampakis There is also a prefix "ocm"

**Comment by tampakis** [comment]
Updated.

**Pull request merged**